### PR TITLE
update sift, dsift, and phow

### PR DESCRIPTION
**Pull request by hubutui**
1. update document
2. update tests
3. remove unneeded function declaration in pxd files
4. code reformat in pxd files
5. declarations should not be declared inline in pxd files, remove
   ``inline``
6. use `scipy.misc.ascent` as test image

Now, `quickshift` still use "lena" as test image. We could remove it soon.

### CODE REVIEW

File: cyvlfeat/sift/dsift.py [hunk 1]
```diff
@@ -158,8 +161,8 @@ def dsift(image, step=1, size=3, bounds=None, window_size=-1, norm=False,
             raise ValueError('Step must only contain positive integers.')
 
     # Validate window_size
-    if not isinstance(window_size, int):
-        raise ValueError('Window size must be an integer.')
+    # if not isinstance(window_size, int):
```
Comment (patricksnape) [inline cyvlfeat/sift/dsift.py:164]: Just delete this?

File: cyvlfeat/sift/phow.py [hunk 2]
```diff
@@ -178,35 +175,35 @@ def phow(image, verbose=False, fast=True, sizes=(4, 6, 8, 10), step=2, color='gr
...
-
-        dim2 = contrast.shape[0]
-        param2 = (sizes[si]) * np.ones((dim2, 1))
-        temp_frames.append(np.append(f[0], param2, axis=1))
-        frames = np.concatenate(temp_frames, axis=0)
-
-        temp_descrs.append(d[0])
-        descriptors = np.concatenate(temp_descrs, axis=0)
+            d[k][contrast < contrast_threshold] = 0
+            # for i, val in enumerate(contrast):
```
Comment (patricksnape) [inline cyvlfeat/sift/phow.py:196]: Delete this?

File: cyvlfeat/sift/phow.py [hunk 3]
```diff
@@ -178,35 +175,35 @@ def phow(image, verbose=False, fast=True, sizes=(4, 6, 8, 10), step=2, color='gr
...
+            #     if val < contrast_threshold:
+            #         d[k][i] = 0
+
+        temp_frames.append(np.column_stack([f[0][:, :3], sizes[si] * np.ones((f[0].shape[0], ), dtype=np.float32)]))
+        temp_descrs.append(np.column_stack(d))
+        # dim2 = contrast.shape[0]
```
Comment (patricksnape) [inline cyvlfeat/sift/phow.py:202]: Remove commented out?

**Comment by patricksnape** [comment]
Yes would be good to remove Lena as soon as possible. Amazing work as always - minor comments just about deleting commented out code

**Comment by patricksnape** [comment]
We can make a release after this is merged?

File: cyvlfeat/sift/dsift.py (hunk 1 above)
Comment (hubutui) [reply cyvlfeat/sift/dsift.py:164]: yes, according to https://github.com/vlfeat/vlfeat/blob/master/vl/dsift.h#L97, the dtype of `windows_size` is `double`.

File: cyvlfeat/sift/phow.py (hunk 3 above)
Comment (hubutui) [reply cyvlfeat/sift/phow.py:202]: ok

File: cyvlfeat/sift/phow.py (hunk 2 above)
Comment (hubutui) [reply cyvlfeat/sift/phow.py]: `phow` implement is converted from the MATLAB code. I rewrite some of the code to avoid redundant for-loop.

**Comment by hubutui** [comment]
@patricksnape To remove `lena` image from cyvlfeat, we still need to update `quickshift`.

**Pull request merged**